wlroots, libinput and propagatedBuildInputs

### Describe the bug

`wlroots` has header files that contains `#include <libinput.h>`. When I try to package software ([wxrd](https://gitlab.freedesktop.org/xrdesktop/wxrd)) based on `wlroots` it errors with:

```
wxrd> /nix/store/ywn7q2np8awinb4j0w90bh52n9gfbf9f-wlroots-0.15.1/include/wlr/backend/libinput.h:12:10: fatal error: libinput.h: No such file or directory
wxrd>    12 | #include <libinput.h>
wxrd>       |          ^~~~~~~~~~~~
```

Even though the application itself isn't referring to `libinput` anywhere in its codebase.

I have a hunch `wlroots` needs to have `libinput` in `propagatedBuildInputs`, but I'm unsure. I see `propagatedBuildInputs` being used very sparingly, but is this a case where it should be used?

Alternatively, should `wxrd` add `libinput` to its `buildInputs` and add it to the `.meson` files as a dependency so that `-I${libinput}/include` is added to cflags?

### Steps To Reproduce
Steps to reproduce the behavior:

```
$ nix build -L github:bobvanderlinden/nixos-config/7db2cc60b93dd23fb7593b014a945a3087f95db7#wxrd
...
wxrd> /nix/store/ywn7q2np8awinb4j0w90bh52n9gfbf9f-wlroots-0.15.1/include/wlr/backend/libinput.h:12:10: fatal error: libinput.h: No such file or directory
wxrd>    12 | #include <libinput.h>
wxrd>       |          ^~~~~~~~~~~~
...
```

### Expected behavior

I expected `libinput.h` being automatically available in environments when `wlroots` is included if `wlroots` refers to `libinput.h` in its header files.

### Screenshots
If applicable, add screenshots to help explain your problem.

### Additional context
Add any other context about the problem here.

### Notify maintainers

@primeos

--- a/pkgs/development/libraries/wlroots/default.nix
+++ b/pkgs/development/libraries/wlroots/default.nix
@@ -79,12 +79,16 @@ let
       ]
       ++ extraNativeBuildInputs;
 
+      propagatedBuildInputs = [
+        # The headers of wlroots #include <libinput.h>, and consumers of `wlroots` need not add it explicitly, hence we propagate it.
+        libinput
+      ];
+
       buildInputs = [
         libliftoff
         libdisplay-info
         libGL
         libcap
-        libinput
         libxkbcommon
         libgbm
         pixman

